ast/src/asm_analysis/mod.rs

```diff
@@ -659,6 +659,8 @@ pub struct SubmachineDeclaration {
     pub name: String,
     /// the type of the submachine
     pub ty: AbsoluteSymbolPath,
+    /// machine arguments
+    pub param_values: Vec<Expression>,
```

Comment:
Why not call it arguments?

Reply:
i was using `params` (for the declaration) and `args` (for the passed values), but i thought it could be confusing as both names can be used interchangeably (GPT somewhat agreed), so I opted to use `values`.

Reply:
I think gpt is wrong, in my opinion it is more confusing to use param_values for arguments...